### ISSUE TITLE
Double Marshaling removes comment

### ISSUE THREAD

**Issue by nettrino** [go-yaml/yaml#532]
When executing the following PoC

```golang
package main

import (
	"fmt"
	"log"

	"gopkg.in/yaml.v3"
)

var (
	sourceYaml = `
x: y
## comment 1

## comment 2

## comment 3

## comment 4
a: z
`
)

func main() {
	yamlNode := yaml.Node{}

	err := yaml.Unmarshal([]byte(sourceYaml), &yamlNode)
	if err != nil {
		log.Fatalf("error: %v", err)
	}

	out, err := yaml.Marshal(&yamlNode)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(string(out))
	fmt.Println("--------------------------")
	out2, err := yaml.Marshal(&yamlNode)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(string(out2))
}
```

the output (using go version go1.13.3) is 
```
x: y
## comment 1

## comment 2

## comment 3

## comment 4
a: z

--------------------------
x: y
## comment 2

## comment 3

## comment 4
a: z
```

Is a deep copy required by the user or is that a bug?

**Comment by niemeyer** [comment]
As described that would certainly be a bug.

I will look into it. Thanks for reporting.

**Comment by niemeyer** [comment]
That's fixed with 0b21df4. The testing added also ensures there are no similar cases lurking.

Thanks again for reporting.
